code/__HELPERS/text.dm

```diff
@@ -257,6 +252,39 @@
 #undef LETTERS_DETECTED
 
 
+/// Much more permissive version of reject_bad_name().
+/// Returns a trimmed string or null if the name is invalid.
+/// Allows most characters except for IC chat prohibited words.
+/proc/permissive_sanitize_name(value)
+	if(!istext(value)) // Not a string
+		return
+
+	var/name_length = length(value)
+	if(name_length < 3) // Too short
+		return
+
+	if(name_length > 3 * MAX_NAME_LEN) // Bad input
+		return
```

Comment:
Is this even necessary if we're going to trim it down to a value shorter than this anyway?

Reply:
Meh, probably not, but this is just copy pasted from `reject_bad_name` which has this comment
https://github.com/tgstation/tgstation/blob/6d544980591b2d04cea13d476a3d96e9328c0451/code/__HELPERS/text.dm#L171-L175